mpr-research-data.py

```diff
@@ -161,34 +122,156 @@ def sliceAndPushToGCP(courseDF, retrieveDF, client, targetBucketName=targetBucke
     return True
 
 
-# RETRIEVE KEYS
-# --------------------------------------------------------------------------
+class Config:
+    def __init__(self):
+        self.targetBucketName: str = 'mpr-research-data-uploads'
+        self.numberOfMonths: int = 4
+        self.defaultQueryFolder: str = 'queries'
+        self.queryTemplateDict: str = {'course': 'courseQuery.sql',
+                                       'retrieve': 'retrieveQuery.sql'}
+        self.dbParams: str = None
+        self.gcpParams: str = None
 
-dbParams = getDBCreds()
-gcpParams = getGCPCreds()
+    def set(self, name, value):
+        if name in self.__dict__:
+            self.name = value
+        else:
+            raise NameError('Name not accepted in set() method')
 
+    def setAndVerifySQLFile(self, queryType):
+        try:
+            self.queryTemplateDict[queryType] = str(
+                os.getenv(f'{queryType.upper()}_QUERY', self.queryTemplateDict[queryType]))
+
+            if self.queryTemplateDict[queryType].lower().endswith('.sql'):
+                logging.info(f'Loading in SQL file - {self.queryTemplateDict[queryType]}')
+            if not os.path.isfile(os.path.join(self.defaultQueryFolder, self.queryTemplateDict[queryType])):
+                logging.error(
+                    f'SQL Query file for {queryType} not found in query directory {self.defaultQueryFolder}.')
+                return False
+            return True
+        except Exception as e:
+            logging.error(f'Error Message: {e}')
+            logging.error(
+                f'Invalid parameter passed for query type: {queryType}.')
+            return False
+
+    def setAndVerifyDBCreds(self):
+
+        dbCredsDefaultDict = {'NAME': None,
+                              'USER': None,
+                              'PASSWORD': None,
+                              'HOST': None,
+                              'PORT': 3306}
+        self.dbParams = {}
+
+        allKeyPartsFound = True
+        for credPart in dbCredsDefaultDict:
+            try:
+                self.dbParams[credPart] = str(os.getenv(
+                    'DB_' + credPart, dbCredsDefaultDict[credPart]))
+            except Exception as e:
+                logging.error(f'Error Message: {e}')
+                logging.error(
+                    f'Invalid parameter passed for DB_{credPart}.')
+                allKeyPartsFound = False
+
+            if not self.dbParams[credPart]:
+                logging.error(
+                    f'Did not find .env variable for M-Write Peer Review production DB key: {credPart}.')
```

Comment:
In production, these variables won't be set by a file called `.env`.  They will just be variables set in the OpenShift environment.  (Maybe from a file, but from within the environment, it won't seem to be.)  It's probably better to say "configuration item" rather than ".env variable".

Reply:
Noted. I will change that.

Reply:
✅